### PR TITLE
Drop SupportedFeatures from gateway-api

### PR DESCRIPTION
**Pull request by howardjohn**
This field provides no value but causes a lot of harm:
* https://github.com/kubernetes-sigs/gateway-api/pull/3200#discussion_r1681853821
* https://github.com/istio/istio/issues/50851

And it looks like there is no conformance requirement to support it.

If we do, we need to make it off-by-default with a flag; I see no reason to provide this.

### CODE REVIEW

**Review by zirain — APPROVED** [review]

**Comment by LiorLieberman** [comment]
> And it looks like there is no conformance requirement to support it.

The plan is to base all conformance tests based on the features in this field that the implementation claims to support. So if it goes in, it will essentially be need to report and run conformance itself.

That said, I added my feedback on https://github.com/kubernetes-sigs/gateway-api/pull/3200, so it may be that we pause it for now. We still need feedback from more folks, as this feature has attracted a lot of interest previously.

**Comment by LiorLieberman** [comment]
Also, a nit, if this is merged, I'd include a release note. I believe some users already have the features populated on their GWC.

**Review by keithmattix — APPROVED** [review]
Approving but +1 to needing a release note

File: pilot/pkg/config/kube/gateway/supported_features.go [hunk 1]
```diff
@@ -15,20 +15,7 @@
 package gateway
 
 import (
-	k8sv1 "sigs.k8s.io/gateway-api/apis/v1"
 	"sigs.k8s.io/gateway-api/pkg/features"
-
-	"istio.io/istio/pkg/util/sets"
 )
 
 var SupportedFeatures = features.AllFeatures
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply pilot/pkg/config/kube/gateway/supported_features.go:21]: this is needed fo the tests, just not for writijg to status

**Pull request merged**